lib/ecto/enum.ex

```diff
@@ -46,17 +50,31 @@ defmodule Ecto.Enum do
   def init(opts) do
     values = Keyword.get(opts, :values, nil)
 
-    unless is_list(values) and Enum.all?(values, &is_atom/1) do
+    unless (is_list(values) and
+              Enum.all?(values, &is_atom/1)) or
+             (Keyword.keyword?(values) and Enum.all?(Keyword.values(values), &is_integer/1)) do
```

Comment:
I believe these chained conditions might compromise a bit the code reading, so I'd suggest something a bit different instead:
```suggestion
    list_of_atoms? = &(is_list(&1) && Enum.all?(&1, &is_atom/1))
    keywords_of_integers? = &(Keyword.keyword?(&1) && Enum.all?(Keyword.values(&1), &is_integer/1))
    
    unless list_of_atoms?.(values) or keywords_of_integers?.(values) do
```